### PR TITLE
QuickStart: Create a virtual network

### PR DESCRIPTION
**Pull request by KumudD**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by KumudD** [comment]
#sign-off @bmoore-msft

**Comment by bmoore-msft** [comment]
@KumudD - check the pipeline results for the BP errors...

**Comment by KumudD** [comment]
@bmoore-msft addressed errors. Thanks!

#sign-off

File: 101-virtual-network-2vms-create/metadata.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "type": "QuickStart",
+    "itemDisplayName": "Create an Azure virtual network",
+    "description": "This template creates an Azure virtual network with two Windows Server 2016 servers in the subnet",
+    "summary": "Deploy a virtual network.  Two virtual machines will be deployed to the single subnet in the virtual network",
+    "githubUsername": "KumudD",
+    "dateUpdated": "2020-06-17",
+    "environments": [
+    "AzureCloud"
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/metadata.json:10]: Why not AzureGov?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,464 @@
+{
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:1]: format the JSON, e.g. SHIFT+ALT+F in VS Code

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+                            "sourceAddressPrefix": "*",
+                            "destinationAddressPrefix": "*",
+                            "access": "Allow",
+                            "priority": 1000,
+                            "direction": "Inbound"
+
+                        }
+                    },
+                    {
+                        "name": "[concat(parameters('networkSecurityGroups_myVm1_name'), '5985')]",
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:106]: why is this port open?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+                            "priority": 1001,
+                            "direction": "Inbound"
+
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Network/networkSecurityGroups",
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:123]: if the NSGs are identical, remove one of them and reuse the other

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('VMNIC2'))]"
+                        }
+                    ]
+                }
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:318]: why is this deployed twice?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+                "destinationPortRange": "3389",
+                "sourceAddressPrefix": "*",
+                "destinationAddressPrefix": "*",
+                "access": "Allow",
+                "priority": 1000,
+                "direction": "Inbound"
+
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:337]: why is this deployed twice?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+                "destinationPortRange": "5985",
+                "sourceAddressPrefix": "*",
+                "destinationAddressPrefix": "*",
+                "access": "Allow",
+                "priority": 1001,
+                "direction": "Inbound"
+
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:356]: why is this deployed twice?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+                "destinationPortRange": "3389",
+                "sourceAddressPrefix": "*",
+                "destinationAddressPrefix": "*",
+                "access": "Allow",
+                "priority": 1000,
+                "direction": "Inbound"
+
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:375]: why is this deployed twice?

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+                                "id": "[variables('subnetRef')]"
+                            },
+                            "primary": true,
+                            "privateIPAddressVersion": "IPv4"
+                        }
+                    }
+                ],
+                "enableAcceleratedNetworking": false,
+                "enableIPForwarding": false,
+                "networkSecurityGroup": {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:423]: move the NSG to the subnet

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+                            "privateEndpointNetworkPolicies": "Enabled",
+                            "privateLinkServiceNetworkPolicies": "Enabled"
+                        }
+                    }
+                ],
+                "enableDdosProtection": false,
+                "enableVmProtection": false
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:214]: put the vm deployments in a copy loop (as well as the nic's and the public IPs) - use arrays for the properties that need to change.

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('vmsize')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "MicrosoftWindowsServer",
+                        "offer": "WindowsServer",
+                        "sku": "2016-Datacenter",
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:230]: use 2019

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,464 @@
...
+                        "version": "latest"
+                    },
+                    "osDisk": {
+                        "osType": "Windows",
+                        "name": "[concat(parameters('virtualMachines_myVm1_name'), '_disk1_aae2e3e988d942d596afcaf64a2e658e')]",
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:235]: remove this property

File: 101-virtual-network-2vms-create/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,464 @@
...
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
```
Comment (bmoore-msft) [inline 101-virtual-network-2vms-create/azuredeploy.json:239]: you need to either use SSD or validate that the VM size support premium disk

**Comment by KumudD** [comment]
@bmoore-msft completed updates as suggested except for the Disk type on the VM, not sure what exactly needs to change. Please advise. Thanks!

**Comment by KumudD** [comment]
@BhargaviAnnadevara-MSFT The template mimics the Quickstart scenario for creating a  virtual network.

**Comment by tfitzmac** [comment]
@bmoore-msft - it seems like the main difference between this template and 201-vm-copy-index-loops is the public IP address. Could we add a conditional public IP address to 201-vm-copy-index-loops and reconcile some of the differences with parameters and hard-coded values?

**Comment by bmoore-msft** [comment]
We can certainly make it conditional - though this section: https://docs.microsoft.com/en-us/azure/virtual-network/quick-create-portal#connect-to-a-vm-from-the-internet implies a publicIp as there's nothing connecting the user to the vnet otherwise.

**Comment by KumudD** [comment]
@bmoore-msft @tfitzmac Here is my ARM template Quickstart [staged doc link](https://review.docs.microsoft.com/en-us/azure/virtual-network/quick-create-template?branch=pr-en-us-119729) that does have a "Connect to the VMN from the internet" section.

**Comment by tfitzmac** [comment]
@KumudD @bmoore-msft - yes, I was thinking the article would enable the public IP address, but if someone was using the existing 201-vm-copy-index-loops and did not want to deploy the public IP address, they could turn it off.

**Comment by bmoore-msft** [comment]
Ok, let's use one of the existing samples and make the changes for the publicIp to be optional...

**Comment by tfitzmac** [comment]
I have a template that is almost ready, just want to verify a couple of things. It revises 201-vm-copy-index-loops

**Pull request closed without merging**